huaweicloud/resource_huaweicloud_cce_node_attach.go

```diff
@@ -330,7 +388,86 @@ func resourceCCENodeAttachV3Create(d *schema.ResourceData, meta interface{}) err
 		return fmtp.Errorf("Error adding HuaweiCloud CCE Node: %s", err)
 	}
 
-	return resourceCCENodeV3Update(d, meta)
+	return resourceCCENodeV3Read(d, meta)
+}
+
+func resourceCCENodeAttachV3Update(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	if d.HasChanges("os", "key_pair", "password") {
+		nodeClient, err := config.CceV3Client(GetRegion(d, config))
+		if err != nil {
+			return fmtp.Errorf("Error creating HuaweiCloud CCE client: %s", err)
+		}
+		clusterID := d.Get("cluster_id").(string)
+		resetOpts := nodes.ResetOpts{
+			Kind:       "List",
+			ApiVersion: "v3",
+		}
+
+		resetNode := nodes.ResetNode{
+			NodeID: d.Id(),
+			Spec: nodes.AddNodeSpec{
+				Os:            d.Get("os").(string),
+				Name:          d.Get("name").(string),
+				ServerConfig:  resourceCCENodeAttachV3ServerConfig(d),
+				VolumeConfig:  resourceCCENodeAttachV3VolumeConfig(d),
+				RuntimeConfig: resourceCCENodeAttachV3RuntimeConfig(d),
+				K8sOptions:    resourceCCENodeAttachV3K8sOptions(d),
+				Lifecycle:     resourceCCENodeAttachV3Lifecycle(d),
+			},
+		}
+
+		resetOpts.NodeList = append(resetOpts.NodeList, resetNode)
+
+		logp.Printf("[DEBUG] Reset node Options: %#v", resetOpts)
+		// Add loginSpec here so it wouldn't go in the above log entry
+		var loginSpec nodes.LoginSpec
+		if hasFilledOpt(d, "key_pair") {
+			loginSpec = nodes.LoginSpec{
+				SshKey: d.Get("key_pair").(string),
+			}
+		} else if hasFilledOpt(d, "password") {
+			loginSpec = nodes.LoginSpec{
+				UserPassword: nodes.UserPassword{
+					Username: "root",
+					Password: d.Get("password").(string),
+				},
+			}
+		}
+		resetOpts.NodeList[0].Spec.Login = loginSpec
+
+		s, err := nodes.Reset(nodeClient, clusterID, resetOpts).ExtractAddNode()
+
+		if err != nil {
+			return fmtp.Errorf("Error resetting HuaweiCloud Node: %s", err)
+		}
+
+		nodeID, err := getResourceIDFromJob(nodeClient, s.JobID, "CreateNode", "InstallNode",
+			d.Timeout(schema.TimeoutUpdate))
+
+		if err != nil {
+			return err
+		}
+		d.SetId(nodeID)
+
+		stateConf := &resource.StateChangeConf{
+			Pending:      []string{"Build", "Installing"},
+			Target:       []string{"Active"},
+			Refresh:      waitForCceNodeActive(nodeClient, clusterID, nodeID),
+			Timeout:      d.Timeout(schema.TimeoutUpdate),
+			Delay:        20 * time.Second,
+			PollInterval: 20 * time.Second,
+		}
+		_, err = stateConf.WaitForState()
+		if err != nil {
+			return fmtp.Errorf("Error resetting HuaweiCloud CCE Node: %s", err)
+		}
+
+		return resourceCCENodeV3Read(d, config)
+
+	} else {
+		return resourceCCENodeV3Update(d, config)
```

Comment:
how about **os** and **tags** are both changed?

Reply:
if os is changed, the node will be reset, all parameters of the node will be set again